[DirectX] Lower `@llvm.dx.typedBufferStore` to DXIL ops (#104253)
llvm/llvm-project: llvm/docs/DirectX/DXILResources.rst

     - ``i32``
     - Index into the buffer

Texture and Typed Buffer Stores
-------------------------------

*relevant types: Textures and TypedBuffer*

> The `TextureStore`_ and `BufferStore`_ DXIL operations can only be used to

Reviewer: ```suggestion 
The `TextureStore`_ and `BufferStore`_ DXIL operations are only allowed to
```

> write all 4 32-bit elements to a texture or a typed buffer. Note that both of

Reviewer: ```suggestion
 write all four 32-bit elements (components) at once to a texture or a typed buffer.
```

> these operations do have a mask parameter, but it is documented that it must

Reviewer: ```suggestion
While both operations include a mask parameter, it is specified that the mask must
```

> cover all components for these types.

Reviewer: ```suggestion
always cover all components when used with these types.
```
